attestation-agent/attestation-agent/src/lib.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::str::FromStr;
+
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use attester::{detect_tee_type, BoxedAttester};
+
+pub mod config;
+mod token;
+
+use token::*;
+
+use crate::config::{aa_kbc_params, Config};
+
+/// Attestation Agent (AA for short) is a rust library crate for attestation procedure
+/// in confidential containers. It provides kinds of service APIs related to attestation,
+/// including the following
+/// - `get_token`: get attestation token from remote services, e.g. attestation services.
+/// - `get_evidence`: get hardware TEE signed evidence due to given runtime_data, s.t.
+/// report data.
+/// - `extend_runtime_measurement`: extend the runtime measurement. This will extend the
+/// current hardware runtime measurement register (if any) or PCR for (v)TPM platforms
+/// with a runtime event.
```

Comment:
note: there's no vTPM impl for that endpoint (yet). I expect there will be one, but I'm not 100% sure yet.

Reply:
Let me add the comments for that